pkg/operator/defragcontroller/defragcontroller.go

```diff
@@ -140,24 +140,22 @@ func (c *DefragController) checkDefrag(ctx context.Context, recorder events.Reco
 		member, err := getMemberFromStatus(etcdMembers, status)
 		if err != nil {
 			errors = append(errors, err)
+			continue
 		}
 
 		// Check each member's status which includes the db size on disk "DbSize" and the db size in use "DbSizeInUse"
 		// compare the % difference and if that difference is over the max diff threshold and also above the minimum
 		// db size we defrag the members state file. In the case where this command only partially completed controller
 		// can clean that up on the next sync. Having the db sizes slightly different is not a problem in itself.
 		if isEndpointBackendFragmented(member, status) {
-			recorder.Eventf("DefragControllerDefragmentAttempt", "Attempting defrag on member: %s, dbSize: %d , dbInUse: %d, memberID %d, leader ID: %d", member.Name, status.DbSize, status.DbSizeInUse, member.ID, status.Leader)
+			recorder.Eventf("DefragControllerDefragmentAttempt", "Attempting defrag on member: %s, memberID: %d, dbSize: %d, dbInUse: %d, leader ID: %d", member.Name, member.ID, status.DbSize, status.DbSizeInUse, status.Leader)
 			if _, err := c.etcdClient.Defragment(ctx, member); err != nil {
 				// defrag can timeout if defragmentation takes longer than etcdcli.DefragDialTimeout
 				errors = append(errors, fmt.Errorf("failed to defragment etcd member: %q :%v", member.Name, err))
-			}
-			postDefragEndpointStatus, err := c.etcdClient.Status(ctx, member)
```

Comment:
this call was right after defrag while the data it added to the event was useful we expose the results via metrics. So instead of complicating the logical flow, I am leaning towards a straiter path.